packages/babel-plugin/src/properties/resolve.js

```diff
@@ -27,11 +47,38 @@ export default function resolveProperty({ types: t, template }) {
     return t.stringLiteral(importArg.node.value)
   }
 
-  return ({ callPath, funcPath }) =>
-    t.objectMethod(
+  /**
+   * @type {import('../index').PropertyFactory}
+   */
+  const factory = (
+    /**
+     * @type {import('../index').PropertyFactoryOptions}
+     */
+    { callPath, funcPath, path },
+  ) => {
+    const properties = path.get('arguments.1.properties')
+
+    const isFederated =
+      moduleFederation ||
+      (Array.isArray(properties) &&
+        properties.some(
+          prop =>
+            prop.isObjectProperty() &&
+            prop.get('key').isIdentifier({ name: 'federated' }) &&
```

Comment:
Not sure how this work and when it should be used.

Please keep in mind that in no circumstances we can use `federated` template on the client-side bundle - that will effectively disable code splitting.
Right now, there is a big probability that it will not happen.

Reply:
Yeah, make sense, the idea now is to use `moduleFederation: true` option only on server-side compiler